### PR TITLE
test(registry): drop third-party contract tests

### PR DESCRIPTION
**Pull request by topher-lo**
## Summary

- remove 15 complete third-party provider-contract suites plus two mocked AWS dispatch tests (138 white-box tests) that mocked vendor behavior without verifying live contracts
- replace a Microsoft Sentinel-backed parser fixture with a synthetic template while preserving Tracecat's date-like-string regression coverage
- retain Kubernetes platform-boundary coverage for credential isolation, blocked file/subprocess credential sources, context selection, ambient-loader prevention, and explicit isolated-client injection; remove only its provider-contract assertions
- preserve provider-neutral registry/template validation, live OpenLDAP coverage, and AWS credential-isolation/AssumeRole security coverage
- codify thin-wrapper guidance for future or materially expanded integrations in `packages/tracecat-registry/AGENTS.md`
- add `packages/tracecat-registry/CLAUDE.md` as a symlink to `AGENTS.md`

This PR makes no runtime integration or public API changes. Existing integration contracts remain authoritative until an explicit migration is planned.

## Third-party integration policy

The registry guidance now requires endpoint-specific research against official documentation, OpenAPI specifications when available, and official SDK or MCP schemas. It prefers direct YAML REST templates, requires a planning choice before adopting an official Python SDK, and keeps integrations close to pass-through wrappers.

The policy also covers API-native inputs, secrets and OAuth, base-URL precedence, encoded path segments, raw HTTP response envelopes, bounded pagination, negative polling predicates, narrow permitted boundary handling, and the prohibition on provider-specific white-box tests. Live or sandbox coverage remains an explicit planning choice when a reliable environment exists.

The test exception is limited to Tracecat-owned platform security or protocol boundaries—for example credential isolation, preventing host filesystem or subprocess access, blocking ambient credential discovery, network-target restrictions, and shared protocol machinery. Provider-local dispatch, validation, pagination, and serialization do not qualify merely because Tracecat implements them.

## Current integration-conformance audit

> [!NOTE]
> This snapshot is advisory, non-exhaustive, and intentionally out of scope for remediation in this PR. Counts describe the current catalog; they do not imply that every historical implementation is defective. Existing public inputs and outputs must not be changed without an explicitly planned migration.

- **Output envelopes:** of 681 YAML tool templates, 517 return `.result.data` and 160 return the full result envelope. The remaining templates use other return shapes.
- **Inline processing:** 199 templates contain `core.script.run_python`, and 17 use `core.transform.*`. These need case-by-case review because small mechanical request assembly remains permitted.
- **Polling:** two body-based polling predicates use exact equality. GitLab's `status_code != 202` predicate illustrates the new negative/transient-state rule.
- **Documentation depth:** 118 templates use catalog-level documentation URLs rather than links to the exact vendor endpoint.
- **Base URLs:** 545 templates declare `base_url`, while 312 use the input-to-workspace-variable fallback.
- **Representative SDK/UDF review candidates:** Freshservice decodes responses into custom success/body shapes and aggregates pagination metadata; Okta normalizes configuration, constructs provider request models, unwraps SDK results, and aggregates pagination; Slack selects SDK response data and includes semantic helpers such as `found`/`not_found` aggregation; Cloudflare validates dispatch names and flattens SDK page objects. Some of these behaviors may be justified as credentials/security, SDK serialization, or bounded pagination, but each should be assessed against the new thin-wrapper rules when materially changed.

## Verification

- `PG_PORT=5832 uv run pytest -q -r fE tests/registry/test_templates.py tests/registry/test_kubernetes_sdk.py tests/unit/test_aws_assume_role.py -n auto` — 706 passed
- `PG_PORT=5832 uv run pytest -q tests/unit/test_aws_assume_role.py tests/unit/test_dsl_common.py -n auto` — 16 passed
- `uv run basedpyright tests/unit/test_dsl_common.py` — 0 errors, 0 warnings
- `uv run ruff check .`
- `uv run ruff format --check .`
- `git diff --check`
- `readlink packages/tracecat-registry/CLAUDE.md` — `AGENTS.md`
- confirmed all 15 selected provider-contract suites are absent, along with the two mocked AWS dispatch tests
- confirmed generic template validation, live LDAP, AWS security, and Kubernetes platform-boundary coverage remain
- confirmed the diff contains no third-party integration implementation changes

## LOC breakdown

| Category | + | - |
| --- | ---: | ---: |
| Tests | 45 | 4,078 |
| Docs | 65 | 0 |

### CODE REVIEW

File: packages/tracecat-registry/AGENTS.md [hunk 1]
```diff
@@ -2,6 +2,68 @@
...
+
+- Do not add provider-specific white-box tests that mock an API or SDK and assert
+  URLs, arguments, schemas, enums, payloads, or outputs. These restate the
+  implementation without validating the real provider contract.
+- Add live or sandbox provider tests only when a reliable environment exists and the
+  user explicitly chooses that coverage during planning.
+- Generic registry and template validation remains required. Narrow unit tests are
+  allowed for Tracecat-owned security or protocol mechanics such as credential
+  isolation, private-method blocking, serialization limits, and reusable pagination
+  machinery.
```
[the remark this answers is not in the document]
Comment (topher-lo) [reply packages/tracecat-registry/AGENTS.md]: Addressed in 34c84abf6. Restored all 13 Kubernetes boundary cases covering file-backed and dynamic credential rejection, context isolation, ambient credential-loader blocking, and explicit injection of the isolated API client. The public-wrapper test now uses a synthetic SDK class and no longer asserts provider arguments or output.

Cloudflare remains removed: its dispatch, pagination, and serialization behavior is provider-local rather than a Tracecat platform trust boundary. The AGENTS guidance now makes that distinction explicit. The retained template, Kubernetes, and AWS coverage passes: 708 tests.
